src/cloud/integrations/gitlab-integration.md

```diff
@@ -0,0 +1,161 @@
+
+   ```terminal
+   origin git@gitlab.com:<user-name>/<repo-name>.git (fetch)
+   origin git@gitlab.com:<user-name>/<repo-name>.git (push)
+   ```
+   {:.no-copy}
+
+1. Push the project files to your new GitLab repository. Remember to keep all branch names the same.
+
+   ```bash
+   git push -u origin master
+   ```
+
+   If you are starting with a new GitLab repository, you may have to use the `-f` option, because the remote repository does not match your local copy.
+
+1. Verify that your GitLab repository contains all of your project files.
+
+## Enable the GitLab integration
+
+The following enables the GitLab integration and provides a Payload URL to use when creating a webhook.
+
+{:.bs-callout-warning}
+The following command overwrites _all_ code in your {{site.data.var.ece}} project with code from your GitLab repository. This includes all branches, including the Production branch. This action happens instantly and cannot be undone. As a best practice, it is very important to clone all of your branches from your {{site.data.var.ece}} project and push them to your GitLab repository **before** adding the GitLab integration.
+
+{:.procedure}
+To enable the GitLab integration:
+
+1. Enable the integration.
+
+   ```bash
+   magento-cloud integration:add --type=gitlab --project=<project-ID> --token=<your-GitLab-token> [--base-url=<GitLab-url> --server-project=<GitLab-project> --build-merge-requests={true|false} --merge-requests-clone-parent-data={true|false} --fetch-branches={true|false} --prune-branches={true|false}]
+   ```
+
+   -  `<project-ID>`—Your {{site.data.var.ece}} project ID
+   -  `<your-GitLab-token>`—The personal access token you generated for GitLab
+   -  `--base-url`-URL of GitLab (https://gitlab.com/ if GitLab is used in its SaaS version)
+   -  `--server-project`-Project name in GitLab (part after the base url)
+   -  `--build-merge-requests`-—An _optional_ parameter that instructs {{site.data.var.ece}} to build a new environment for each and every merge requests (`true` by default)
+   -  `--merge-requests-clone-parent-data`-—An _optional_ parameter that instructs {{site.data.var.ece}} to clone the parent environment's data for merge requests (`true` by default)
+   -  `--fetch-branches`—An _optional_ parameter that causes {{site.data.var.ece}} to fetch all branches from the remote (as inactive environments) (`true` by default)
+   -  `--prune-branches`—An _optional_ parameter that instructs {{site.data.var.ece}} to delete branches that do not exist on the remote (`true` by default)
+
+   **Example**: Enable the GitLab integration for a private repository:
+
+   ```bash
+   magento-cloud integration:add --type gitlab --project=3txxjf32gtryos --token=qVUfeEn4ouze7A7JH --base-url=https://gitlab.com/ --server-project=my-agency/project-name --build-merge-requests=false --merge-requests-clone-parent-data=false --fetch-branches=true --prune-branches=true
+   ```
+
+1. Enter the required information if prompted.
```

Comment:
```suggestion
1. When prompted, enter "Y" to add the integration.
   
    ```terminal
    Warning: adding a 'gitlab' integration will automatically synchronize code from the external Git repository.
   This means it can overwrite all the code in your project.

    Are you sure you want to continue? [y/N] y
```

Reply:
OK